src/python/pants/option/arg_splitter.py

```diff
@@ -84,7 +84,7 @@ class ArgSplitter:
     def __init__(self, known_scope_infos: Iterable[ScopeInfo], buildroot: str) -> None:
         self._buildroot = Path(buildroot)
         self._known_scope_infos = known_scope_infos
-        self._known_scopes = {si.scope for si in known_scope_infos} | {
+        self._known_scopes = {si.scope for si in known_scope_infos if si.is_goal} | {
```

Comment:
this is the culprit that both solves the initial issue, but also introduces alot of broken tests in options_test.py..

Reply:
Yeah, good attempt, but I think the filtering needs to be elsewhere. This generates the "universe of known scopes" for all options parsing. I'll take a closer look.

Reply:
@benjyw cool. I'll let you take a look at it. I just spent 5 more minutes on it, after getting a night of sleep ;)
so, got rid of the test issues I had, and now it behaves properly for `./pants pex` etc. Not sure if there are some hidden corner cases I'm not aware of, though..